### PR TITLE
Connection quality APIs

### PR DESCRIPTION
**Pull request by davidzhao**
See server PR: https://github.com/livekit/livekit-server/pull/167

### CODE REVIEW

**Review by davidliu — APPROVED** [review]
LGTM

**Review by hiroshihorie — APPROVED** [review]
🚀

**Comment by jibon57** [comment]
This solution is better then mine. Not sure if it possible to show download & upload speed as well as. This will be helpful for user to check their internet speed.

File: src/room/Room.ts [hunk 1]
```diff
@@ -116,6 +117,8 @@ class Room extends EventEmitter {
       this.state = RoomState.Connected;
       this.emit(RoomEvent.Reconnected);
     });
+
+    this.engine.on(EngineEvent.ConnectionQualityUpdate, this.handleConnectionQualityUpdate);
```
Comment (lukasIO) [inline src/room/Room.ts:121]: Wouldn't this effectively override any quality setting that has been set manually?
In our usecase we need to decide the quality not only on connection speed but also depending on other parameters.

**Pull request merged**